lib/packwerk/constant_discovery.rb

```diff
@@ -52,10 +63,10 @@ def context_for(const_name, current_namespace_path: [])
 
       package = @packages.package_from_path(constant.location)
       ConstantContext.new(
-        constant.name,
-        constant.location,
-        package,
-        package&.public_path?(constant.location),
```

Comment:
Note that this save navigation is removed since the package is considered non-nilable.

I could still add this back with `T.unsafe(package)&.public_path`, but in various other parts of the codebase we expect the package to be non-nilable (wherever we call `constant.package`). Those places would error if the package did not exist.

When we get the package using `package_from_path`, we default to `root` if another package cannot exist, so this plus the fact that other parts would break makes me think this is okay.